src/components/atoms/UserAvatar/UserAvatar.tsx

```diff
@@ -59,13 +60,7 @@ export const _UserAvatar: React.FC<UserAvatarProps> = ({
     [`UserAvatar--${size}`]: size,
   });
 
-  // @debt this seems to cause performance issues. It should be possible to use worldUserLocationsById
-  //   from useWorldUserLocation / similar, plus some 'is recent user?' logic to directly look up the individual
-  //   user in an efficient manner.
-  const isOnline = useMemo(
-    () => recentWorldUsers.find((worldUser) => worldUser.id === user?.id),
-    [user, recentWorldUsers]
-  );
+  const isOnline = useMemo(() => !isEmpty(userLastSeenIn), [userLastSeenIn]);
```

Comment:
You could push this logic back into the useWorldUserLocation hook so that it can be used elsewhere. It could also potentially make sense to have some ‘is recent user’ sort of logic in there as well potentially?

Reply:
> You could push this logic back into the useWorldUserLocation hook so that it can be used elsewhere.
Makes sense, will do in the next performance updates as well as addressing the other `TODOs`.

> It could also potentially make sense to have some ‘is recent user’ sort of logic in there as well potentially?
I'll just expose `isOnline` from `useWorldUserLocation`